newIDE/app/src/InstancesEditor/index.js

```diff
@@ -736,14 +746,36 @@ export default class InstancesEditor extends Component<Props> {
       !this.hasCursorMovedSinceItIsDown &&
       this.keyboardShortcuts.shouldCloneInstances()
     ) {
-      this.hasCursorMovedSinceItIsDown = true;
-
       const selectedInstances = this.props.instancesSelection.getSelectedInstances();
-      for (var i = 0; i < selectedInstances.length; i++) {
-        const instance = selectedInstances[i];
-        this.props.initialInstances
-          .insertInitialInstance(instance)
-          .resetPersistentUuid();
+
+      // Current cursor position
+      const cursorPosX = this.getLastCursorSceneCoordinates()[0];
+      const cursorPosY = this.getLastCursorSceneCoordinates()[1];
+      if (!this.duplicateInstanceLastCursorRetrieved) {
+        // Initial cursor position
+        const cursorSceneCoords = this.getLastCursorSceneCoordinates();
+        this.duplicateInstanceLastCursorX = cursorSceneCoords[0];
+        this.duplicateInstanceLastCursorY = cursorSceneCoords[1];
+        this.duplicateInstanceLastCursorRetrieved = true;
+      }
+
+      // Compare current cursor position with initial position
+      // If offset is enough, we create the instance
+      if (
+        cursorPosX - this.duplicateInstanceLastCursorX > DUPLICATE_OFFSET ||
+        cursorPosX - this.duplicateInstanceLastCursorX < -DUPLICATE_OFFSET ||
+        cursorPosY - this.duplicateInstanceLastCursorY > DUPLICATE_OFFSET ||
+        cursorPosY - this.duplicateInstanceLastCursorY < -DUPLICATE_OFFSET
+      ) {
+        this.hasCursorMovedSinceItIsDown = true;
+        for (var i = 0; i < selectedInstances.length; i++) {
+          const instance = selectedInstances[i];
+          this.props.initialInstances
+            .insertInitialInstance(instance)
+            .resetPersistentUuid();
+        }
+      } else {
+        return;
```

Comment:
No need for this `else return;`.

What you could do though, is do this:
```js
if (!distantEnough) return;
this.hasCursorMovedSinceItIsDown = true;
...
```
This way, you don't get an additionnal indentation level.